Lib/http/server.py

```diff
@@ -737,7 +737,16 @@ def send_head(self):
 
             # Use HTTP compression (gzip) if possible
             accept_encoding = self.headers.get("Accept-Encoding", "")
-            encodings = [x.strip() for x in accept_encoding.split(",")]
+
+            encodings = []
+            for accepted in accept_encoding.split(","):
+                encoding, *quality = accepted.lower().split(";")
+                if quality:
+                    q, *v = quality[0].split("=")
+                    if q == "q" and v and v[0] == "0":
```

Comment:
In a7f1f47 I followed your suggestion to use `split_header_words()`. I'm not sure _http.cookiejar_ is the most natural place to put a headers parser, but it's another topic, maybe for another discussion on the issue tracker.